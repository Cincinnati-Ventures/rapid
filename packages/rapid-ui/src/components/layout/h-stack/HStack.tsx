import React from 'react';
import Stack from '../Stack/Stack';

type Spacing = 'sm' | 'md' | 'lg';

interface VStackProps
	extends React.DetailedHTMLProps<
		React.HTMLAttributes<HTMLDivElement>,
		HTMLDivElement
	> {
	styles?: string;
	spacing?: Spacing;
}

const HStack = React.forwardRef<HTMLDivElement, VStackProps>(
	({ styles, children, spacing = 'lg', ...rest }, ref) => {
		// This component does not support custom dividers or wrapping (maybe a TODO ?)
		return (
			<Stack
				{...rest}
				ref={ref}
				styles={styles}
				spacing={spacing}
				direction='row'
			>
				{children}
			</Stack>
		);
	},
);

export default HStack;
