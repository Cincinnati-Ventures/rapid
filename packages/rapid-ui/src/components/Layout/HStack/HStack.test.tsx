import React from 'react';
import { render } from '@testing-library/react';
import { expect, it, describe } from 'vitest';
import HStack from './HStack';

describe('HStack', () => {
	it('should render default correctly', () => {
		const { container } = render(<HStack />);
		expect(container).toMatchSnapshot();
	});

	it('should render correctly with non default props', () => {
		const { container } = render(<HStack styles='p-4' spacing='sm' />);
		expect(container).toMatchSnapshot();
	});

	it('should render correctly with children', () => {
		const { container } = render(
			<HStack styles='bg-red-500' spacing='md'>
				<div>
					<p>Hello world!</p>
				</div>
				<div>
					<p>Hello world!</p>
				</div>
				<div>
					<p>Hello world!</p>
				</div>
				<div>
					<p>Hello world!</p>
				</div>
			</HStack>,
		);
		expect(container).toMatchSnapshot();
	});
});
