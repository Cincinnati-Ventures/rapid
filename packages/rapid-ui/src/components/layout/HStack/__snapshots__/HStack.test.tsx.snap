// Vitest Snapshot v1

exports[`HStack > should render correctly with children 1`] = `
<div>
  <div
    class="flex space-x-3 bg-red-500 rapid-stack"
  >
    <div>
      <p>
        Hello world!
      </p>
    </div>
    <div>
      <p>
        Hello world!
      </p>
    </div>
    <div>
      <p>
        Hello world!
      </p>
    </div>
    <div>
      <p>
        Hello world!
      </p>
    </div>
  </div>
</div>
`;

exports[`HStack > should render correctly with non default props 1`] = `
<div>
  <div
    class="flex space-x-1.5 p-4 rapid-stack"
  />
</div>
`;

exports[`HStack > should render default correctly 1`] = `
<div>
  <div
    class="flex space-x-6 rapid-stack"
  />
</div>
`;
